### PR TITLE
TAJO-1798: Dynamic partitioning occasionally fails.

### PR DESCRIPTION
**Pull request by blrunner**
I verified this patch with TPC-H 1G.

### CODE REVIEW

**Comment by blrunner** [comment]
I converted List<PartitionDescProto> to Set<PartitionDescProto> for avoiding duplicated partitions. And after stored partitions to catalog, cleared all Set<PartitionDescProto> instances.

File: tajo-core/src/main/java/org/apache/tajo/querymaster/Query.java [hunk 1]
```diff
@@ -505,30 +510,31 @@ private QueryState finalizeQuery(Query query, QueryCompletedEvent event) {
...
-            simpleTableName = desc.getName();
+            LOG.info("Can't find partitions for adding.");
           }
-
-          // Store partitions to CatalogStore using alter table statement.
-          catalog.addPartitions(databaseName, simpleTableName, partitions, true);
-        } else {
-          LOG.info("Can't find partitions for adding.");
+          query.clearPartitions();
+          partitions.clear();
```
Comment (jinossy) [inline tajo-core/src/main/java/org/apache/tajo/querymaster/Query.java]: please remove unnecessary code

**Comment by jinossy** [comment]
Could you add tests ?

**Comment by blrunner** [comment]
Thanks @jinossy 
I've just updated the patch using your comments.

**Comment by hyunsik** [comment]
Your change mainly appears to focus on partition bugs. Only this line (https://github.com/apache/tajo/pull/709/files#diff-f6cacc8f5447d717968f06d0b133f171L530) is related to the issue title. So, this issue would be more searchable if this issue title is changed into a better one.

**Comment by blrunner** [comment]
Thanks @hyunsik 
I've just changed the title. :)

**Comment by jinossy** [comment]
+1 LGTM

**Comment by blrunner** [comment]
Thanks @jinossy  and @hyunsik 
I've just committed it to master branch and 0.11.0 branch.

**Pull request closed without merging**